Hibernate/OneToOneMapping/Main.java

```diff
@@ -0,0 +1,34 @@
+package OneToOneMapping;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+
+
+public class Main {
+
+	public static void main(String[] args) {
+		SessionFactory factory = new Configuration()
+				.configure("hibernate.cfg.xml")
+				.addAnnotatedClass(Instructor.class)
+				.addAnnotatedClass(InstructorDetail.class)
+				.buildSessionFactory();
+
+		Session session = factory.getCurrentSession();
+		try {	
+			Instructor tempInstructor = new Instructor("Joe", "Darby", "Joe@gmail.com");
+			InstructorDetail tempInstructorDetail =new InstructorDetail("http://www.joeofficial.com/youtube",
+							"Singing!!!");	
+			tempInstructor.setInstructorDetail(tempInstructorDetail);
+			session.beginTransaction();
+			System.out.println("Saving instructor: " + tempInstructor);
```

Comment:
I think you forgot to call session.save
Call it and see what happens when you just save the instructor object without saving instructor detail ?